### DISCUSSION TITLE
Spotube doesn't let me play in background like Spotify

### DISCUSSION THREAD

**Discussion by ghost** [KRTirtho/spotube#878] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The "notification player" closes as soon as Spotube is closed.

### Expected Behavior

The "notification player" should NOT close despite of closing the app, and keep playing music even after Spotube is closed. There would be an X symbol to close the notification player (or this is how Spotify Lite works).

### Steps to reproduce

1. Open Spotube
2. Play any track
3. Close Spotube. The music stops as the app is completely shut down

### Operating System

Android 12

### Spotube version

V3.2

### Installation source

F-Droid (Android)

**Comment by ksharizard** [comment]
You need to disable battery optimisations for spotube. This will allow it to run in background and play your music. Go to settings -> apps and notifications -> spotube -> disable battery optimisations.

**Reply by ghost** [reply]
I have Battery Optimisation turned off. :)
The issue still persists.

@KRTirtho Please fix this...

**Reply by ghost** [reply]
Just for reference, the free _Metro_ app, which is a local music player, is able to play in the background EVEN WITH BATTERY OPTIMISATION TURNED ON.

So, I think Spotube should be able to do so too... background playback is an important factor for a music client 🙄

App referenced:
https://www.f-droid.org/en/packages/io.github.muntashirakon.Music/

Metro notification player
![Screenshot_20231115-191217~2.png](https://github.com/KRTirtho/spotube/assets/85873175/7dbc10ee-726e-4216-92af-a4cf01abdfa7)

(Android level UI, Metro's default, also the same in Spotify Lite)

![Screenshot_20231115-191231~2.png](https://github.com/KRTirtho/spotube/assets/85873175/7fe8d8be-300f-462f-b097-0b60f7e62298)

(Metro Classic Notification UI)

**Reply by KRTirtho** [reply]
What do you mean by "closing the app"? Do you mean exiting the app or clearing the app from recent apps?

**Reply by ghost** [reply]
Clearing the app from Recents...

**Reply by KRTirtho** [reply, marked as answer]
Yes, in that case, it is expected to stop playing. Some local audio players do offer that but Spotube works differently. Because it's cross-platform it doesn't the native audio player that android provides.

**Reply by ghost** [reply]
Wouldn't it be possible to add that in a future update?

The problem is, I am accustomed to the muscle memory of clearing the apps from Recents because music players like Spotify Lite, Metro, etc. work fine with it...

Perhaps need some time to adapt to Spotube 🙃
I'm not going back to Spotify because the new restrictions are too much.. can't even seek on Spotify!

**Reply by KRTirtho** [reply]
Spotify official app also stops playing as soon as app is cleared. If you have a habit of clearing, try locking the app in the recent apps. It'll not get cleared then

**Reply by ghost** [reply]
Oh yes, I'm sorry. You are right, Spotify Lite does that, not Spotify.
